File: src/Shared/Logistics.Shared.Consts/SalaryType.cs
using System.ComponentModel;

namespace Logistics.Shared.Consts;

public enum SalaryType
{
    [Description("None")]
    None,
    
    [Description("Monthly")]
    Monthly,
    
    [Description("Weekly")]
    Weekly,
    
    [Description("Share of gross")]
    ShareOfGross
}

File: src/Client/Logistics.OfficeApp/src/app/core/enums/salary-type.ts
import {EnumLike, EnumType, convertEnumToArray, findValueFromEnum} from "./enum-like";

export enum SalaryType {
  None,
  Monthly,
  Weekly,
  ShareOfGross,
}

export const SalaryTypeEnum: EnumLike = {
  None: {value: 0, description: "None"},
  Monthly: {value: 1, description: "Monthly"},
  Weekly: {value: 2, description: "Weekly"},
  ShareOfGross: {value: 3, description: "Share of gross"},

  getValue(enumValue: string | number): EnumType {
    return findValueFromEnum(this, enumValue);
  },

  toArray(): EnumType[] {
    return convertEnumToArray(this);
  },
};
